model/riscv_pmp_control.sail

```diff
@@ -95,7 +84,7 @@ function pmpMatchEntry(addr: physaddr, width: xlenbits, acc: AccessType(ext_acce
   match pmpMatchAddr(addr, width, rng) {
     PMP_NoMatch      => PMP_Continue,
     PMP_PartialMatch => PMP_Fail,
-    PMP_Match        => if   pmpCheckPerms(ent, acc, priv)
+    PMP_Match        => if pmpCheckRWX(ent, acc) | (priv == Machine & not(pmpLocked(ent)))
```

Comment:
I guess the other options would be 
1) pass priv to pmpcheckRWX() and add `if priv == Machine & not(pmpLocked(ent)) then true else match ...`?
2) change pmpCheckPerms to an if instead of a match().

But 2) seems unnecessary since there is only one caller of pmpCheckPerms() so we might as inline it. LGTM.